### PR TITLE
Add automatic config search

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: model_analyzer/config/input/config.py [hunk 1]
```diff
@@ -255,10 +254,53 @@ def objective_list_output_mapper(objectives):
...
+            ConfigField(
+                'perf_analyzer_cpu_util',
+                field_type=ConfigPrimitive(float),
+                default_value=40,
+                description=
+                "Maximum CPU utilization value allowed for the perf_analyzer.")
+        )
+        self._add_config(
+            ConfigField(
+                'run_config_search_max_concurrency',
```
Comment (aramesh7) [inline model_analyzer/config/input/config.py:275]: Do you think we can make these a config section? Like `run_config_search_parameters`

File: model_analyzer/config/run/run_config_generator.py [hunk 2]
```diff
@@ -107,51 +125,178 @@ def _generate_model_config_combinations(self, value):
         # always return a list.
         return [value]
 
-    def _generate_run_configs(self):
+    def execute_run_configs(self):
```
Comment (aramesh7) [inline model_analyzer/config/run/run_config_generator.py:128]: Do you think we can extract out a class called `RunConfigExecutor` for this? This config generationcan happen in the current class but the execution may be able to be moved

File: model_analyzer/config/run/run_config_generator.py [hunk 3]
```diff
@@ -107,51 +125,178 @@ def _generate_model_config_combinations(self, value):
...
+        triton_launch_mode = self._analyzer_config['triton_launch_mode']
+
+        measurements = {}
+        while self._run_configs:
+            # Remove one run config from the list
+            run_config = self._run_configs.pop()
+
+            model_config = run_config.model_config()
+            original_model_name = run_config.model_name()
+            model_name = model_config.get_field('name')
```
Comment (aramesh7) [inline model_analyzer/config/run/run_config_generator.py:144]: can we call this `model_config_name`?

File: model_analyzer/config/run/run_config_generator.py [hunk 4]
```diff
@@ -107,51 +125,178 @@ def _generate_model_config_combinations(self, value):
...
                 model.model_name(), model)
+
+            # Add the new run config.
             self._run_configs.append(
                 RunConfig(model.model_name(), model_config, perf_configs))
 
+    def _generate_run_configs(self):
+        model = self._model
+        analyzer_config = self._analyzer_config
+        triton_launch_model = analyzer_config['triton_launch_mode']
```
Comment (aramesh7) [inline model_analyzer/config/run/run_config_generator.py:246]: possible typo? `triton_launch_model` -> `triton_launch_mode`

File: model_analyzer/config/run/run_search.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        Returns true if the amount of throughput gained
+        is reasonable for continuing the search process
+        """
+
+        # If number of measurements is smaller than 4,
+        # the search can continue.
+        if len(self._measurements) < 4:
+            return True
+
+        if self._calculate_throughput_gain(1) > 0.05 or \
```
Comment (aramesh7) [inline model_analyzer/config/run/run_search.py:123]: Could you add 0.05 to `model_analyzer/constants.py` and import it here?

**Review by aramesh7 — CHANGES_REQUESTED** [review]

File: model_analyzer/config/run/run_config_generator.py (hunk 2 above)
Comment (Tabrizian) [reply model_analyzer/config/run/run_config_generator.py:128]: I'll refactor this later. It takes a long time to do this.

File: model_analyzer/config/input/config.py (hunk 1 above)
Comment (Tabrizian) [reply model_analyzer/config/input/config.py:275]: We may want to add this CLI and it will be easier to interact with them if they are at the top level.

**Review by aramesh7 — APPROVED** [review]

**Pull request merged**